base/strings/annotated.jl

```diff
@@ -399,6 +399,51 @@ Get all annotations of `chr`, in the form of a vector of annotation pairs.
 """
 annotations(c::AnnotatedChar) = c.annotations
 
+## Character transformation helper function, c.f. `unicode.jl`.
+
+"""
+    annotated_chartransform(str::AnnotatedString, f::Function, state=nothing)
```

Comment:
Bikeshedding: this seems almost like a `map(f, str)` function, which would suggest `f` should be first argument.

The it could also be used in a `do`-block, e.g. for `titlecase`, and then you can perhaps also get rid of `state`.

That said, as I understand it, this is an internal function, so feel free to 100% ignore me on this :-)

Reply:
I wanted to keep `f` and `state` together, but the `do` syntax is good motivation. I think I will swap this around.